lib/dns.js

```diff
@@ -61,17 +61,16 @@ function makeAsync(callback) {
       // The API already returned, we can invoke the callback immediately.
       callback.apply(null, arguments);
     } else {
-      process.nextTick(callMakeAsyncCbNT, callback, arguments);
+      var args = new Array(arguments.length + 1);
+      args[0] = callback;
```

Comment:
It seems a little bit hacky.
The following code is easy to understand for me.
BUT, I have not bench which code is faster.

```javascript
var args = [callback];
for (var i = 0; i <arguments.length; i++)
  args.push(arguments[i]);
process.nextTick.apply(null, args);
```